### DISCUSSION TITLE
Importing Data and Interpolation

### DISCUSSION THREAD

**Discussion by Rahul-GeoPhysics** [gempy-project/gempy#626] [General]
Hi, I am having issues while loading my csv file into Gempy. I have X,Y,Z, Density values in the spreadsheet, that I want to load in, interpolate, grid and then export as another csv (that can store the interpolated values). But when I am trying to load in the data using the following commands and trying to see the surfaces, it doesn't show anything.

**import gempy as gp
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import os
geo_model=gp.create_model("model_1")
gp.init_data(geo_model,[-124.8,-120.0,40.0,49.2,-101000.0,3000.0],[28,55,15],path="density_from_Brocher_2005_meters.csv",default_values=True)
geo_model.surfaces**

<img width="361" alt="image" src="https://user-images.githubusercontent.com/76798266/125849494-04ac1e40-95b6-42b1-a8de-dca3db1c1f5a.png">

**Comment by javoha** [comment]
Hello @Rahul-GeoPhysics,

can you share your csv file so I can take a direct look. that would be easiest if possible.

**Reply by Rahul-GeoPhysics** [reply]
@javoha Here it is:  
[rho.csv](https://github.com/cgre-aachen/gempy/files/6849456/rho.csv)
I am trying to interpolate the x,y,z,rho values and export as a csv. Tried using scipy, but didn't quite find it useful

**Reply by javoha** [reply]
Thanks, so I see you have nearly 100,000 data points. This is too much for gempy at the moment. In its core these universal cokriging approaches work best with sparse data. Also can you explain why you want to use gempy and the potential field method here instead of classical geostatistical interpolation (e.g kriging)?

**Reply by Rahul-GeoPhysics** [reply]
Hi @javoha, my aim is to create a density model with x,y,z,rho values in the csv file. The issues are: the dataset has missing coordinates and rho values that require 3d interpolation. I am not sure if there is anything else I can try. Do you have any suggestion?

Thanks for the reply!

**Reply by javoha** [reply]
I think you are looking for a different method. Gempy is made to create a strucutral geologic model based on surface boundaries (generally the boundary between two rock units). While you can think of your density thresholds as boundaries, Gempy requires only points exactly on these surfaces as input (because that is what you would map in t typical borehole log or outcrop), not data points in between. If I interpret your dataset correctly, you have a densely sampled set of density measurements. With such a dense set you could probably get awy with doing some kind of inverse distance or nearest neighbour interpolation, but if you wanna be a little more fancy, classical kriging seems to be the way to go. Do you have any experience with that. GSTools is a nice open source tool for that, but you need to find a suitable variogram model and some geostatistics backgorund might help.
